Lib/pickle.py

```diff
@@ -202,10 +202,8 @@ def commit_frame(self, force=False):
             f = self.current_frame
             if f.tell() >= self._FRAME_SIZE_TARGET or force:
                 with f.getbuffer() as data:
-                    n = len(data)
                     write = self.file_write
-                    write(FRAME)
-                    write(pack("<Q", n))
+                    write(FRAME + pack("<Q", len(data)))
```

Comment:
Can you please add a short comment explaining this change? At the first read, this change seems inefficient.